buildengine/discover.go

```diff
@@ -7,40 +7,72 @@ import (
 	"path/filepath"
 	"sort"
 
-	"github.com/TBD54566975/ftl/common/moduleconfig"
+	"github.com/TBD54566975/ftl/internal/log"
 )
 
-// DiscoverModules recursively loads all modules under the given directories.
+// DiscoverProjects recursivley loads all modules under the given directories
+// (or if none provided, the current working directory is used) and external
+// libraries in externalLibDirs.
+func DiscoverProjects(ctx context.Context, dirs []string, externalLibDirs []string, stopOnError bool) ([]Project, error) {
```

Comment:
```suggestion
func DiscoverProjects(ctx context.Context, moduleDirs []string, externalLibDirs []string, stopOnError bool) ([]Project, error) {
```

Just to make it a bit clearer.